llvm/lib/Target/AArch64/AArch64InstrAtomics.td

```diff
@@ -543,6 +543,20 @@ let Predicates = [HasLSE] in {
   defm : LDOPregister_patterns_mod<"LDCLR", "atomic_load_and", "ORN">;
 }
 
+defm atomic_load_fadd  : binary_atomic_op_fp<atomic_load_fadd>;
+defm atomic_load_fmin  : binary_atomic_op_fp<atomic_load_fmin>;
+defm atomic_load_fmax  : binary_atomic_op_fp<atomic_load_fmax>;
+
+let Predicates = [HasLSFE] in {
+  defm : LDFPOPregister_patterns<"LDFADD",   "atomic_load_fadd">;
+  defm : LDFPOPregister_patterns<"LDFMAXNM", "atomic_load_fmax">;
+  defm : LDFPOPregister_patterns<"LDFMINNM", "atomic_load_fmin">;
+
+  defm : LDBFPOPregister_patterns<"LDBFADD",   "atomic_load_fadd">;
+  defm : LDBFPOPregister_patterns<"LDBFMAXNM", "atomic_load_fmax">;
+  defm : LDBFPOPregister_patterns<"LDBFMINNM", "atomic_load_fmin">;
```

Comment:
Should these also be guarded by HasSVEB16B16?

Reply:
No, I don't think so. If you look at `AArch64InstrInfo.td`, the `LDBFADDA` and similar instructions are guarded only by `[HasLSFE]`. (They were originally also guarded by `HasSVEB16B16`, before we upstreamed them, but we removed that last September).

Reply:
You're right, I can see that these are no longer guarded by the HasSVEB16B16 [here](https://developer.arm.com/documentation/ddi0602/2024-12/SIMD-FP-Instructions/LDBFADD--LDBFADDA--LDBFADDAL--LDBFADDL--BFloat16-floating-point-add-in-memory-?lang=en).
Sorry for the confusion!